### ISSUE TITLE
Feature Request: Add review order (from settings) to reviews screen

### ISSUE THREAD

**Issue by jeff8v7** [davidsansome/tsurukame#340]
The review order setting is great, but I find myself frequently switching from the reviews screen to the settings screen and back to the reviews screen. 

It would be much more convenient if the review order could be set from the reviews screen itself.

For example, let's say I want to do all of the current level (most important) reviews first, then I want to do the oldest (easiest) reviews first, then I want to finish what's left in SRS order.

Or, let's say I'm doing my reviews, and I get to one I can't quite remember... I want to randomize the order to get around it, and pick it up at the end of my session. 

Or any other crazy idea which requires tweaking the reviews order while I'm in the middle of my reviews...

**Comment by turtlemaster19** [comment]
For clarification, when you say reviews screen, that would be the menu that includes the Wrap-Up button.

**Comment by jeff8v7** [comment]
That would be a perfect place for it :-)

**Comment by jeff8v7** [comment]
As I get deeper into reviews, I'm noticing more desire (opportunity?) to affect the review order. 

My ideal lofty goal would be that each review session starts with a "review order" screen - showing all items which are eligible for review.  A user could then sort and filter them with a few taps.  For example, I could sort by "radicals, kanji, vocab", sub-sort by "highest level", then "randomize" etc. (or whatever weird preference I have at the moment).  Or maybe I could filter for "radicals only" for this review session.

The same screen could be reachable via "Sort / Filter" in the menu which includes the Wrap-Up button.

(I don't know if it's helpful to suggest such things... but I can't help but dream... lol)

**Comment by turtlemaster19** [comment]
@jeff8v7 Okay, you're kidding, right? Showing **all** items available **before** reviewing them?!? Why on earth would you want to do that?

**Comment by jeff8v7** [comment]
The goal is just to preview the order. just show the written items.

Definitely wouldn't show the meaning or reading, or it would give away all the answers...

**Comment by jeff8v7** [comment]
think of this like the "lesson filter" user script, but adapted for reviews. I probably need to do a mockup for this idea to make sense...

**Comment by turtlemaster19** [comment]
No, it makes sense now. For the sorting in general then, see #201 and #304. This request, then, must primarily be to be able to change the review order that already exists without leaving reviews.

---

There then seems to be an optional request to view all of the items in advance without the meaning/reading that will come in a review. 

(Personally, I don't see why I would want to know... it would only make me want to look some up, but to each their own.)

**Comment by jeff8v7** [comment]
> This request, then, must primarily be to be able to change the review order that already exists without leaving reviews.

Yes - would be convenient

> optional request to view all of the items in advance without the meaning/reading that will come in a review.

Right, should probably be a separate request. The idea is a review "preview screen", which allows filtering & sorting, and has a way to visualize how that filter will effect the review session. Like the review summary screen but at the beginning.

I can make it a separate request, or just forget it :-)

**Comment by jeff8v7** [comment]
> it would only make me want to look some up

Ah! I finally got your point. If you know what reviews are coming, you might cheat and look them up beforehand. (Never thought of that)

Ok... scratch that 2nd idea/request.

**Comment by turtlemaster19** [comment]
> Or, let's say I'm doing my reviews, and I get to one I can't quite remember... I want to randomize the order to get around it, and pick it up at the end of my session.

By the way, if you want to do that now, use the TestFlight build, and turn on "Allow skipping reviews"

**Comment by turtlemaster19** [comment]
![image](https://user-images.githubusercontent.com/46784000/81948860-cf780000-95c7-11ea-9cbc-0c559feb22e9.png)

@jeff8v7 Here it is in action!

**Comment by jeff8v7** [comment]
Awesome! The "Review item type order" is a nice touch too. Super convenient here. Thanks!
